src/utils/clearUserStates.ts

```diff
@@ -1,10 +1,14 @@
 import * as Sentry from '@sentry/react'
 import { Dispatch } from '@reduxjs/toolkit'
+import { resetFarmsUserData } from 'state/farms'
+import { resetPoolsUserData } from 'state/pools'
+import { clearAllTransactions } from 'state/transactions/actions'
 import { connectorLocalStorageKey } from '@pancakeswap/uikit'
 import { connectorsByName } from './web3React'
-import { clearAllTransactions } from '../state/transactions/actions'
 
 export const clearUserStates = (dispatch: Dispatch<any>, chainId: number) => {
+  dispatch(resetFarmsUserData())
+  dispatch(resetPoolsUserData())
```

Comment:
I think instead of cleaning redux state separately, we could centralize the logic by capturing the logout action and cleaning everything we need. We could do it in root reducer or add a higher order reducer "cleanupReducer" to achieve this

Reply:
Done